components/inputs/input-number.js

```diff
@@ -63,14 +89,24 @@ class InputNumber extends LitElement {
 				class="d2l-input"
 				?disabled="${this.disabled}"
 				id="${this._inputId}"
-				max="${ifDefined(this.max)}"
-				min="${ifDefined(this.min)}"
 				name="${ifDefined(this.name)}"
 				placeholder="${ifDefined(this.placeholder)}"
-				step="${ifDefined(this.step)}"
-				type="number"
-				.value="${this.value}">
+				type="text"
+				.value="${this._formattedValue}">
 		`;
 	}
+
+	_handleChange(e) {
+		const inputValue = e.target.value.trim();
+		this.value = inputValue !== '' ? parseNumber(inputValue) : undefined;
+
+		if (!isNaN(this.value) && this.value !== undefined) {
+			this.value = checkMinMaxValue(this.value, this.min, this.max);
+			this._formattedValue = formatValue(this.value, this.minFractionDigits, this.maxFractionDigits);
+			e.target.value = this._formattedValue;
+		} else {
+			e.target.value = '';
+		}
```

Comment:
One piece you could add here (in this PR or a future one) is to fire `change` event if that actual value changes as a result of the change from the input.